hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java

```diff
@@ -128,16 +124,16 @@ public static void checkMapConfig(MapConfig mapConfig,
                                       SplitBrainMergePolicyProvider mergePolicyProvider,
                                       HazelcastProperties properties) {
 
-        checkNotNativeWhenOpenSource(mapConfig.getInMemoryFormat());
```

Comment:
How come we don't check this anymore when the config is parsed through `MemberDomConfigProcessor`?

Reply:
There was a test failure, but I could not remember. I tried run `ConfigValidatorTest`, the only test class that is affected and it is green.
I'll revert the change